StRoot/StFstRawHitMaker/StFstRawHitMaker.cxx

```diff
@@ -390,6 +391,48 @@ Int_t StFstRawHitMaker::Make()
 
     LOG_INFO << " Total number of FST Raw Hits - Step II = " << mFstCollectionPtr->getNumRawHits() << " w/ idTruth = " << nIdTruth_Fst << endm;
 
+    // Fill Fst raw hits we got so far into StEvent if the chain option is called
+    if(IAttr("fstEvtRawHit")) {//True for storing FST Raw hits
+        LOG_INFO << " Filling FST Raw Hits into StEvent" << endm;
+
+        //prepare for StEvent
+        mEvent = (StEvent *) GetInputDS("StEvent");
+
+        if(mEvent) {
+            LOG_DEBUG<<"Found StEvent"<<endm;
+        } else {
+            mEvent = new StEvent();
+            AddData(mEvent);
+            LOG_DEBUG <<"Added StEvent"<<endm;
+        }
+
+        // Get pointer to an existing StFstEvtCollection if any
+        mFstEvtCollection = mEvent->fstEvtCollection();
+
+        // If no fst raw hit collection, create one
+        if (!mFstEvtCollection) {
+            mFstEvtCollection = new StFstEvtCollection();
+            mEvent->setFstEvtCollection(mFstEvtCollection);
+            LOG_DEBUG << "Make() - Added new StFstEvtCollection to this StEvent" << endm;
+        }
+
+        if(mFstCollectionPtr->getNumRawHits() > 0) {
+            for(int wedgeIdx=0; wedgeIdx<kFstNumWedges; ++wedgeIdx ){
+                StFstRawHitCollection *rawHitCollectionPtr = mFstCollectionPtr->getRawHitCollection( wedgeIdx );
+                if( rawHitCollectionPtr ){
+                    std::vector<StFstRawHit*>& rawHitVec = rawHitCollectionPtr->getRawHitVec();
+                    std::vector< StFstRawHit* >::iterator rawHitIter;
+
+                    for( rawHitIter = rawHitVec.begin(); rawHitIter != rawHitVec.end(); ++rawHitIter ){
+                        StFstRawHit* rawHit = *rawHitIter;
+                        StFstRawHit* newRawHit = new StFstRawHit( *rawHit );
+                        mFstEvtCollection->addRawHit(newRawHit);
+                    }
```

Comment:
This copying of allocated StFstRawHit objects from one vector to another is very questionable... but at least do it in one line like so:

```suggestion
                    //#include <algorithm>
                    std::transform(begin(rawHitVec), end(rawHitVec), back_inserter(mFstEvtCollection->rawHits()), [](const StFstRawHit* h) { return new StFstRawHit(*h);} );
```

Reply:
I have tried your suggestion, however, I got errors as follow:
```
/usr/include/c++/4.8.2/bits/stl_iterator.h:436:7: error: no type named 'value_type' in 'class StSPtrVecFstRawHit'
       operator=(const typename _Container::value_type& __value)
       ^
/usr/include/c++/4.8.2/bits/stl_iterator.h:443:7: error: no type named 'value_type' in 'class StSPtrVecFstRawHit'
       operator=(typename _Container::value_type&& __value)
       ^
In file included from /usr/include/c++/4.8.2/algorithm:62:0,
                 from /afs/rhic.bnl.gov/star/ROOT/5.34.38/.sl73_gcc485/rootdeb/include/TMath.h:34,
                 from /afs/rhic.bnl.gov/star/packages/DEV/StRoot/StChain/StMaker.h:16,
                 from /afs/rhic.bnl.gov/star/packages/DEV/StRoot/StChain/StRTSBaseMaker.h:12,
                 from .sl73_gcc485/obj/StRoot/StFstRawHitMaker/StFstRawHitMaker.h:12,
                 from .sl73_gcc485/obj/StRoot/StFstRawHitMaker/StFstRawHitMaker.cxx:1:
/usr/include/c++/4.8.2/bits/stl_algo.h: In instantiation of '_OIter std::transform(_IIter, _IIter, _OIter, _UnaryOperation) [with _IIter = __gnu_cxx::__normal_iterator<StFstRawHit**, std::vector<StFstRawHit*> >; _OIter = std::back_insert_iterator<StSPtrVecFstRawHit>; _UnaryOperation = StFstRawHitMaker::Make()::__lambda0]':
.sl73_gcc485/obj/StRoot/StFstRawHitMaker/StFstRawHitMaker.cxx:432:171:   required from here
/usr/include/c++/4.8.2/bits/stl_algo.h:4926:12: error: no match for 'operator=' (operand types are 'std::back_insert_iterator<StSPtrVecFstRawHit>' and 'StFstRawHit*')
  *__result = __unary_op(*__first);
            ^
/usr/include/c++/4.8.2/bits/stl_algo.h:4926:12: note: candidates are:
In file included from /usr/include/c++/4.8.2/bits/stl_algobase.h:67:0,
                 from /usr/include/c++/4.8.2/bits/char_traits.h:39,
                 from /usr/include/c++/4.8.2/string:40,
                 from /usr/include/c++/4.8.2/stdexcept:39,
                 from /usr/include/c++/4.8.2/array:38,
                 from .sl73_gcc485/obj/StRoot/StFstRawHitMaker/StFstRawHitMaker.h:5,
                 from .sl73_gcc485/obj/StRoot/StFstRawHitMaker/StFstRawHitMaker.cxx:1:
/usr/include/c++/4.8.2/bits/stl_iterator.h:402:11: note: std::back_insert_iterator<StSPtrVecFstRawHit>& std::back_insert_iterator<StSPtrVecFstRawHit>::operator=(const std::back_insert_iterator<StSPtrVecFstRawHit>&)
     class back_insert_iterator
           ^
/usr/include/c++/4.8.2/bits/stl_iterator.h:402:11: note:   no known conversion for argument 1 from 'StFstRawHit*' to 'const std::back_insert_iterator<StSPtrVecFstRawHit>&'
/usr/include/c++/4.8.2/bits/stl_iterator.h:402:11: note: std::back_insert_iterator<StSPtrVecFstRawHit>& std::back_insert_iterator<StSPtrVecFstRawHit>::operator=(std::back_insert_iterator<StSPtrVecFstRawHit>&&)
/usr/include/c++/4.8.2/bits/stl_iterator.h:402:11: note:   no known conversion for argument 1 from 'StFstRawHit*' to 'std::back_insert_iterator<StSPtrVecFstRawHit>&&'
```
Did I miss anything?

Reply:
Right, seems like the `StSPtrVec...` types wrapping `std::vector` are not very suitable for the standard algorithms